### PR TITLE
IGNITE-22874 Support Transaction aware SQL in thin client

### PR DESCRIPTION
**Pull request by nizhikov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/tx/SqlTransactionsIsolationTest.java [hunk 1]
```diff
@@ -101,7 +103,10 @@ public enum ExecutorType {
         CLIENT,
 
         /** */
-        THIN
+        THIN_VIA_CACHE_API,
```
Comment (alex-plekhanov) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/tx/SqlTransactionsIsolationTest.java:106]: Why thin client is created via private API TcpIgniteClient and not by Ignition.startClient?

Comment (nizhikov) [reply modules/calcite/src/test/java/org/apache/ignite/internal/processors/tx/SqlTransactionsIsolationTest.java:106]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientCache.java [hunk 2]
```diff
@@ -80,13 +80,13 @@
  */
 public class TcpClientCache<K, V> implements ClientCache<K, V> {
     /** "Keep binary" flag mask. */
-    private static final byte KEEP_BINARY_FLAG_MASK = 0x01;
+    public static final byte KEEP_BINARY_FLAG_MASK = 0x01;
```
[the remark this answers is not in the document]
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientCache.java:83]: fixed

File: modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientCache.java [hunk 3]
```diff
@@ -80,13 +80,13 @@
...
 
     /** "Transactional" flag mask. */
-    private static final byte TRANSACTIONAL_FLAG_MASK = 0x02;
+    public static final byte TRANSACTIONAL_FLAG_MASK = 0x02;
```
[the remark this answers is not in the document]
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientCache.java:86]: fixed

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/tx/SqlTransactionsIsolationTest.java [hunk 4]
```diff
@@ -1114,8 +1141,10 @@ public List<List<?>> sql(String sqlText, int[] parts, Object... args) {
         if (!F.isEmpty(parts))
             qry.setPartitions(parts);
 
-        if (type == ExecutorType.THIN)
-            return thinCli.query(qry).getAll();
+        if (type == ExecutorType.THIN_VIA_CACHE_API)
+            return unwrapBinary(thinCli.query(qry).getAll());
+        else if (type == ExecutorType.THIN_VIA_QUERY)
+            return unwrapBinary(thinCli.cache(F.first(thinCli.cacheNames())).query(qry).getAll());
```
[the remark this answers is not in the document]
Comment (nizhikov) [reply modules/calcite/src/test/java/org/apache/ignite/internal/processors/tx/SqlTransactionsIsolationTest.java:1147]: fixed

**Review by alex-plekhanov — APPROVED** [review]

**Pull request merged**